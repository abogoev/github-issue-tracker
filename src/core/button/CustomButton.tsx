import React, { VFC } from "react";
import { StyleSheet, Text } from "react-native";
import theme from "../../theme/theme";
import CustomTouchableOpacity, {
  CustomTouchableOpacityProps,
} from "../touchableopacity/CustomTouchableOpacity";

interface Props extends Omit<CustomTouchableOpacityProps, "children"> {
  title: string;
}

const CustomButton: VFC<Props> = ({ style, title, ...props }) => {
  return (
    <CustomTouchableOpacity style={[styles.container, style]} {...props}>
      <Text style={styles.text}>{title}</Text>
    </CustomTouchableOpacity>
  );
};

const styles = StyleSheet.create({
  container: {
    height: 32,
    backgroundColor: theme.palette.green,
    justifyContent: "center",
    alignItems: "center",
    borderRadius: theme.borderRadius,
  },
  text: {
    ...theme.typography.regularTextBold,
    color: theme.palette.white,
  },
});

export default CustomButton;
