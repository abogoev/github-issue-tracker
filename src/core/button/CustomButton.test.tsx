import React from "react";
import CustomButton from "./CustomButton";
import { create } from "react-test-renderer";

describe("OpenClosedBadge", () => {
  it("should render correctly", () => {
    const component = create(<CustomButton title="Hello" />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it("onPress should be called", () => {
    const mockOnPress = jest.fn();
    const component = create(
      <CustomButton title="Test" onPress={mockOnPress} />
    );
    expect(mockOnPress).not.toBeCalled();

    component.root.props.onPress();
    expect(mockOnPress).toBeCalledTimes(1);
  });
});
