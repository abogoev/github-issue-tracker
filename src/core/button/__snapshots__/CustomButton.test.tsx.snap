// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OpenClosedBadge should render correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgb(45, 164, 78)",
      "borderRadius": 6,
      "height": 32,
      "justifyContent": "center",
      "opacity": 1,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#fff",
        "fontFamily": "Helvetica",
        "fontSize": 14,
        "fontWeight": "500",
        "lineHeight": 20,
      }
    }
  >
    Hello
  </Text>
</View>
`;
